Automattic/wp-calypso
# reduxify user-settings in `me/account`

This PR is partly a spin-off from https://github.com/Automattic/wp-calypso/pull/49105. It doesn't contain certain refactorings (such as migrating `<Account />` to a class component) to make the PR more contained and easier to review. I'll follow up with a few more PRs which take care of those refactorings.

#### Changes proposed in this Pull Request

* Move `<Account />` to use Redux to change / update user settings
* Tangential:
  * fix a bug where changing the language would not protect the user from navigating away
  * fix a bug where changing user settings back to original values wouldn't let the user navigate away
  * fix a bug where saving user settings from the one form would let the user navigate away even though there are changes in the other form

#### Testing instructions

Note Also test this with a newly created account (to test settings which miss a default value such as `colorScheme`) as well as a non-a8c account with a verified email address to be able to change your username (can be the same).

1. Go to `/me/account` and verify updating account and interface settings work exactly like on prod (except for the mentioned bug fixes).
2.  Make sure the following scenarios work as expected:

- Make changes in both forms but only save one of them. Try to navigate away. A dialog should pop up warning you that you have unsaved changes. After you saved the other form, you should be able to navigate away
- Make changes to the form, navigate away (confirm the dialog) and go back. Make sure changes have been cleared and you're back to the initial state.
- Inspect the _Network_ request and verify only settings from the submitted form are being sent
- Check error handling works as expected by blocking requests to `public-api.wordpress.com/rest/v1.1/me/settings`
- Changing language settings should result in a page reload

Related to #24162 
Fixes https://github.com/Automattic/wp-calypso/issues/49591
